### PR TITLE
Check for ApiStatus::kFailure

### PR DESCRIPTION
**Pull request by eyhhuang**

### CODE REVIEW

File: third_party/scsi_mock/engine.cc [hunk 1]
```diff
@@ -71,6 +78,13 @@ ScsiToNvmeResponse ScsiToNvme(unsigned char* cmd_buf, unsigned short cmd_len,
   translator::CompleteResponse cpl_resp =
       translation.Complete(nvme_cpl, buffer_in, sense_buffer);
 
+  if (cpl_resp.status == ApiStatus::kFailure) {
+    Print("Incorrect usage of Translation Library API");
+    ScsiToNvmeResponse resp = {.return_code = 0x40, .alloc_len = 0};
+    translation.AbortPipeline();
```
Comment (Ralith) [inline third_party/scsi_mock/engine.cc:84]: Isn't this already called by `Complete`?

File: third_party/scsi_mock/engine.cc [hunk 2]
```diff
@@ -33,6 +33,13 @@ ScsiToNvmeResponse ScsiToNvme(unsigned char* cmd_buf, unsigned short cmd_len,
   translator::BeginResponse begin_resp =
       translation.Begin(scsi_cmd, buffer, lun);
 
+  if (begin_resp.status == ApiStatus::kFailure) {
+    Print("Incorrect usage of Translation Library API");
+    ScsiToNvmeResponse resp = {.return_code = 0x40, .alloc_len = 0};
+    translation.AbortPipeline();
```
Comment (Ralith) [inline third_party/scsi_mock/engine.cc:39]: Do we actually leave any resources allocated in the case of an early failure?

Comment (eyhhuang) [reply third_party/scsi_mock/engine.cc:39]: Good catch! In Translation::Begin, the only case of a `ApiStatus::kFailure` occurs before any resource allocation.

File: third_party/scsi_mock/engine.cc (hunk 1 above)
Comment (eyhhuang) [reply third_party/scsi_mock/engine.cc:84]: When `Complete` returns an `ApiStatus::kFailure`, it doesn't call `AbortPipeline`. I'm still investigating whether it is necessary to call `AbortPipeline`. In any case, the user (`engine.cc`) probably shouldn't be responsible for calling `AportPipeline`

Comment (Ralith) [reply third_party/scsi_mock/engine.cc:84]: Ah, I see e.g. https://github.com/googleinterns/scsi2nvme/blob/master/lib/translator/translation.cc#L153-L160 could cause a failure return with allocated resources. I agree this should probably be changed to clean up automatically.

**Pull request merged**